### DISCUSSION TITLE
running e2e tests

### DISCUSSION THREAD

**Discussion by iamhaaamed** [obytes/react-native-template-obytes#269] [Q&A] [answered]
Hello,
To run the e2e tests, I have to manually select localhost:8081 in the emulator each time to start running the tests, is there a way to automate this?

**Comment by yjose** [comment]
Did you mean that the Expo Dev Menu shows up on every launch in dev mode? If yes, I think with Expo 50, the app will start with the last dev URL selected, which means that the issue is fixed.

**Reply by rbshh** [reply, marked as answer]
If you're using Metro as your bundler, add the following code to your metro.config.js file to set the default server configuration:

`defaultConfig: {
  server: {
    host: 'localhost',
    port: 8081,
  },
}`

Another possible solution in case of android is to utilise the built-in 'adb' command-line tool:

1. Open a terminal within Android Studio.
2. Start your test runner.
3. Execute the following command to open the emulator's browser at localhost:8081

```bash
adb shell am start -a android.intent.action.VIEW -d 'http://localhost:8081'
```
